biogrid/libs/biogrid-simulator/src/lib/biobrain/biobrain-class.ts

```diff
@@ -12,9 +16,191 @@ export class BioBrain implements Brain {
     }
     return this.instance;
   }
-
+  
   computeAction(state: StateGraph): GridAction {
+
+    let shortestDistances = state.getShortestDistances();
+    
+    this.clonedGraph = state.cloneStateGraph();
+
+    // Create an object of buildings with the energyProviders which supplied
+    // TODO assign the building as Building not the names
+    let buildingSuppliers: SupplyingPath = this.buildingCharging(
+      this.getGridItems()[GRID_ITEM_NAMES.ENERGY_USER],
```

Comment:
seems like we are never using the full list of grid items. Instead - can this.getGridItems take as a parameter the name which you want, and return only those items? 
e.g. this.getGridItems(GRID_ITEM_NAMES.ENERGY_USER) returns a list of all energy users

Reply:
This creates an error since we are returning either BioBattery[] or Building[] or BioEnergySource[]. For instance if we want the getGridItems(GRID_ITEM_NAMES.ENERGY_USER), this can technically not only return the energy users but can also return batteries or solar. This cause the compiler to throw an error because we expected to pass energy users not either energy users or batteries..... I will look at a way to implement it using a helper method but there if I change the method, then I have to getGridItems for every type differently.

Reply:
You can avoid calling `this.getGridItems()` a bunch of times (abbreviated, but you can extrapolate):

```
const {
  [GRID_ITEM_NAMES.ENERGY_USER]: energyUser,    // JS object deconstruction & renaming
  [GRID_ITEM_NAMES.SMALL_BATTERY]: smallBattery,
} = this.getGridItems();

let buildingSuppliers: SupplyingPath = this.buildingCharging(energyUser, smallBattery);
```